first ludo review of group 5 by group 3 and 8

skjohansen opened the pull request.

AndreMorad committed:
@andremorad

AndreMorad committed:
@andremorad

Install entity framework sql server

AndreMorad committed:
@andremorad

added sql serveer to project

singelmamma committed:
@singelmamma

Added classes

AndreMorad committed:
@andremorad

GameEngine project

AndreMorad committed:
@andremorad @singelmamma

player and color

singelmamma committed:
@singelmamma

Created database, added unittest projekt and sorted classes and methods

singelmamma committed:
@singelmamma

Added unit test example

singelmamma committed:
@singelmamma

Added GE method for tokenmovment and some unit tests

singelmamma committed:
@singelmamma

Bluemace committed:
@bluemace

Added documentation + flowchart

Bluemace committed:
@bluemace

Changed name on flowchart

Bluemace committed:
@bluemace

Changed flowchart

simbasang commented on Source/GameEngine/TokenMovement.cs:
```diff
@@ -0,0 +1,42 @@
+
+namespace GameEngine
+{
+    public class TokenMovement
+    {
+        public static int ThrowDie()
+        {
+            Random dieThrow = new Random();
+
+            int result = dieThrow.Next(1, 7);
+
+            return result;
+        }
+
+        public static int MoveToken(int tokenCurrentPosition, int dieResult)
+        {
+            int tokenNewPosition = tokenCurrentPosition;
+
+            for (int i = 1; i <= dieResult; i++)
+            {
+                if (tokenNewPosition >= 45)
+                {
+                    for (int y = i; y <= dieResult; y++)
+                    {
+                        i++;
+                        tokenNewPosition--;
+                    }
+                }
+                else
+                {
+                    tokenNewPosition++;
+                }
+            }
+            return tokenNewPosition;
+        }
+    }
+
+
+}
```
borde inte detta vara metoder i token klassen?

BenYtt commented on Source/TheLudoGame/Program.cs:
```diff
@@ -0,0 +1,49 @@
+    {
+        private static void Main(string[] args)
+        {
+            //Test för att se om databasen lagrar som den ska
+            List<string> colors = new List<string> { "röd", "blå", "grön", "gul" };
+            Game testGame = new Game();
+
+            Console.WriteLine("Hur många spelare?: ");
+            int antalSpelare = int.Parse(Console.ReadLine());
+            for (int i = 1; i <= antalSpelare; i++)
+            {
+                var newPlayer = new Player();
+                Console.WriteLine($"Skriv ditt namn spelare: {i}");
+                newPlayer.PlayerName = Console.ReadLine();
+                Console.WriteLine("Välj färg");
+                for (int x = 0; x < colors.Count; x++)
+                {
+                    Console.WriteLine($"{x} , {colors[x]}");
+                }
+                for (int y = 0; y < 4; y++)
+                {
+                    var newToken = new Token();
+                    newPlayer.Tokens.Add(newToken);
+                }
+                int colorChose = int.Parse(Console.ReadLine());
+                newPlayer.PlayerColor = colors[colorChose];
+                testGame.Players.Add(newPlayer);
+                colors.RemoveAt(colorChose);
+                Console.Clear();
+            }
+
+            var context = new MyContext();
+            context.Add<Game>(testGame);
+            context.SaveChanges();
+            var testDatabas = context.Games.Include(g => g.Players).Where(g => g.Finished == false).FirstOrDefault();
+        }
```
Antar att denna ska in i menyn sen? :)

simbasang commented on Source/TheLudoGame/Database/MyContext.cs:
```diff
@@ -0,0 +1,30 @@
+        public DbSet<Token> Tokens { get; set; }
+        protected override void OnConfiguring(DbContextOptionsBuilder dbContext)
+        {
+            dbContext.UseSqlServer(@"Data Source=den1.mssql8.gear.host;Initial Catalog=dbtheludogame;User id=dbtheludogame;password=Ip5ych-!9Vb1;");
```
hårdkoda inte er connectionstring gör en appseting.json fil istället

Bluemace committed:
@bluemace

Added link to google doc

BenYtt commented on Source/UnitTestGameEngineMethods/UnitTest1.cs:
```diff
@@ -0,0 +1,32 @@
+        public void TestingDieThrow()
+        {
+            int testResult = TokenMovement.ThrowDie();
+            Assert.IsTrue(testResult >= 1 && testResult <= 6);
```
Det enda ni egentligen testar här är ifall Random.Next(1,7) fungerar? Det skulle man ju kunna anta att det gör.

simbasang commented on Source/TheLudoGame/Program.cs:
```diff
@@ -0,0 +1,49 @@
+    {
+        private static void Main(string[] args)
+        {
+            //Test för att se om databasen lagrar som den ska
+            List<string> colors = new List<string> { "röd", "blå", "grön", "gul" };
+            Game testGame = new Game();
+
+            Console.WriteLine("Hur många spelare?: ");
+            int antalSpelare = int.Parse(Console.ReadLine());
+            for (int i = 1; i <= antalSpelare; i++)
+            {
+                var newPlayer = new Player();
+                Console.WriteLine($"Skriv ditt namn spelare: {i}");
+                newPlayer.PlayerName = Console.ReadLine();
+                Console.WriteLine("Välj färg");
+                for (int x = 0; x < colors.Count; x++)
+                {
+                    Console.WriteLine($"{x} , {colors[x]}");
+                }
+                for (int y = 0; y < 4; y++)
+                {
+                    var newToken = new Token();
+                    newPlayer.Tokens.Add(newToken);
+                }
+                int colorChose = int.Parse(Console.ReadLine());
+                newPlayer.PlayerColor = colors[colorChose];
+                testGame.Players.Add(newPlayer);
+                colors.RemoveAt(colorChose);
+                Console.Clear();
+            }
+
+            var context = new MyContext();
+            context.Add<Game>(testGame);
+            context.SaveChanges();
+            var testDatabas = context.Games.Include(g => g.Players).Where(g => g.Finished == false).FirstOrDefault();
+        }
+
```
Detta kan ni bryta ner i mindre methoder

BenYtt commented on Source/UnitTestGameEngineMethods/UnitTest1.cs:
```diff
@@ -0,0 +1,32 @@
+    public class UnitTest1
+    {
+        [TestMethod]
+        public void TestingDieThrow()
+        {
+            int testResult = TokenMovement.ThrowDie();
+            Assert.IsTrue(testResult >= 1 && testResult <= 6);
+        }
+
+        [TestMethod]
+        public void TestingTokenMovmentForward()
+        {
+            int testForwardPosition = TokenMovement.MoveToken(31, 5);
+            Assert.IsTrue(testForwardPosition == 36);
+        }
+
+        [TestMethod]
+        public void TestingTokenMovmentBackwards()
+        {
+            int testBackwardPosition = TokenMovement.MoveToken(43, 5);
+            Assert.IsTrue(testBackwardPosition == 42);
+        }
+
```
Bra att ni gör unit tester!

kaptenen replied:
Är namngivningen korrekt för testmetoderna?

Bluemace committed:
@bluemace

Uppdaterad Länk till rätt dokumentation

BenYtt reviewed (commented):
Bra att ni kommit igång och stilig flowchart!

kaptenen commented on Source/GameEngine/TokenMovement.cs:
```diff
@@ -0,0 +1,42 @@
+
+namespace GameEngine
+{
+    public class TokenMovement
+    {
+        public static int ThrowDie()
+        {
+            Random dieThrow = new Random();
+
+            int result = dieThrow.Next(1, 7);
+
+            return result;
+        }
+
+        public static int MoveToken(int tokenCurrentPosition, int dieResult)
+        {
+            int tokenNewPosition = tokenCurrentPosition;
+
+            for (int i = 1; i <= dieResult; i++)
+            {
+                if (tokenNewPosition >= 45)
+                {
+                    for (int y = i; y <= dieResult; y++)
+                    {
+                        i++;
+                        tokenNewPosition--;
+                    }
+                }
+                else
+                {
+                    tokenNewPosition++;
+                }
+            }
+            return tokenNewPosition;
+        }
```
Borde detta inte vara metoder i Token klassen?

simbasang commented on Source/UnitTestGameEngineMethods/UnitTest1.cs:
```diff
@@ -0,0 +1,32 @@
+    public class UnitTest1
+    {
+        [TestMethod]
+        public void TestingDieThrow()
+        {
+            int testResult = TokenMovement.ThrowDie();
+            Assert.IsTrue(testResult >= 1 && testResult <= 6);
+        }
+
+        [TestMethod]
+        public void TestingTokenMovmentForward()
+        {
+            int testForwardPosition = TokenMovement.MoveToken(31, 5);
+            Assert.IsTrue(testForwardPosition == 36);
+        }
+
+        [TestMethod]
+        public void TestingTokenMovmentBackwards()
+        {
+            int testBackwardPosition = TokenMovement.MoveToken(43, 5);
+            Assert.IsTrue(testBackwardPosition == 42);
```
är inte namngivnigen  metodnamnet_input_önskadoutput

simbasang reviewed (commented):
Ser bra ut känns som ni har en plan med flowcharten!

kaptenen reviewed (commented):
Väldigt bra dokumentation lätt att förstå hur ni vill att programmet skall fungera. Och bra med testerna

Bluemace committed:
@bluemace @andremorad  @singelmamma

Ändrar i kod för omstrukturering

Bluemace committed:
@bluemace @andremorad @singelmamma

Moved all the classes, the database and migrations into "GameEngine" and added "CreatePlayer" Method.

singelmamma committed:
@singelmamma

AndreMorad committed:
@andremorad

reinstall

singelmamma committed:
@singelmamma

Uppdaterat så migration fungerar via GameEngine

singelmamma committed:
@Singelmamma

Added checkwinner method and some unit tests

singelmamma committed:
@Singelmamma

Added new method for player to choose tokens

singelmamma committed:
@singelmamma

Added new method for playerturn

AndreMorad committed:
@andremorad

change in createPlayer

AndreMorad committed:
@andremorad

working with createPlayer

AndreMorad committed:
@andremorad

new organization

singelmamma committed:
@singelmamma

Sorted out a lot, created new parts to the game engine and made a demo game from start to finish for players with one token

singelmamma committed:
@singelmamma

Another update on demogame

singelmamma committed:
@singelmamma @andremorad @bluemace

Bluemace committed:
Added Documentation

singelmamma committed:
@singelmamma

Added comments and deleted DieClass

singelmamma committed:
Merge branch 'master' of https://github.com/PGBSNH19/the-ludo-game-grupp-5

singelmamma committed:
@singelmamma

Updated documentation

singelmamma committed:
@singelmamma

added doc-files

singelmamma committed:
@singelmamma @bluemace @andremorad

singelmamma committed:
@singelmamma

Update method and gitigonre on connectionsstring json-file

singelmamma committed:
Delete appsettings.json

singelmamma committed:
@singelmamma

singelmamma committed:
Delete MyContext.cs

singelmamma committed:
@singelmamma

singelmamma committed:
@singelmamma

Changed name on method

AndreMorad committed:
@andremorad

add xUnit test projekt

singelmamma committed:
@singelmamma

AndreMorad committed:
@andremorad

AndreMorad committed:
@andremorad

AndreMorad committed:
Merge branch 'master' of https://github.com/PGBSNH19/the-ludo-game-grupp-5

singelmamma committed:
@singelmamma

added test

AndreMorad committed:
@andremorad

AndreMorad committed:
Merge branch 'master' of https://github.com/PGBSNH19/the-ludo-game-grupp-5

singelmamma committed:
@singelmamma

Deleted files

singelmamma committed:
@singelmamma

Deleted more files and added tests

singelmamma committed:
@singelmamma

Added som more unit tests and renamned some of the old tests

singelmamma committed:
@singelmamma

Rearranged GameWinner test

AndreMorad committed:
@andremorad

added KnockOutAnotherToken xUnitTest

singelmamma committed:
@singelmamma

AndreMorad committed:
Merge branch 'master' of https://github.com/PGBSNH19/the-ludo-game-grupp-5

singelmamma committed:
@singelmamma

Added intergration testfile and moq package

Bluemace committed:
@bluemace

Unittest for InGoal.

AndreMorad committed:
@andremorad

added TokensToMove unit test

AndreMorad committed:
@andremorad

added Thread.Sleep and moved it to GameMenu

AndreMorad committed:
@andremorad

Changet my test  to pass

Bluemace committed:
@bluemace

Added Tests for UpdateTurnAndRound

singelmamma committed:
@singelmamma

Added new gamemessages and changed methods call in gameflow

singelmamma committed:
@singelmamma

"Token knocked out antoher token with the same color" bugfixed

singelmamma committed:
@singelmamma

Added more unit test

AndreMorad committed:
@andremorad

some changes

singelmamma committed:
@singelmamma

singelmamma committed:
Delete appsettings.dev.json

AndreMorad committed:
@andremorad

AndreMorad committed:
Merge branch 'master' of https://github.com/PGBSNH19/the-ludo-game-grupp-5

singelmamma committed:
@singelmamma

singelmamma committed:
Merge branch 'master' of https://github.com/PGBSNH19/the-ludo-game-grupp-5

singelmamma committed:
@singelmamma

singelmamma committed:
@singelmamma @andremorad  @bluemace

Uppdaterade spelflödet och testerna

singelmamma committed:
@singelmamma @andremorad

Rearranged the tests to different testfiles and added som more tests

AndreMorad committed:
@andremorad

added new test

AndreMorad committed:
@andremorad

added colors to console

singelmamma committed:
@singelmamma

singelmamma committed:
@singelmamma

singelmamma committed:
@singelmamma

singelmamma committed:
@singelmamma @andremorad @bluemace

Reviderade koden och tog bort onödiga metoden, samt ny namngivning på vissa metoder och varibler

singelmamma committed:
@singelmamma

singelmamma committed:
@singelmamma

singelmamma committed:
@singelmamma

singelmamma committed:
@singelmamma @andremorad  @bluemace

singelmamma committed:
@singelmamma

singelmamma committed:
@singelmamma
